src/core/qgspallabeling.cpp

```diff
@@ -2935,6 +2944,55 @@ void QgsPalLayerSettings::parseTextBuffer( QgsRenderContext &context )
   dataDefinedValEval( DDBlendMode, QgsPalLayerSettings::BufferBlendMode, exprVal, context.expressionContext() );
 }
 
+void QgsPalLayerSettings::parseTextMask( QgsRenderContext &context )
+{
+  QVariant exprVal; // value() is repeatedly nulled on data defined evaluation and replaced when successful
```

Comment:
You can return early if

   if ( !mDataDefinedPreopreties.hasActiveProperties() )
      return;

Reply:
I think it is already done in the caller (registerFeature)